### PR TITLE
Refactor permission handling

### LINKED ISSUE #359: Separate personal and admin tables
Eg. On print page, the history table shows all user's history. I think this should be added to the admin's print page, and we should show only the personal history on the normal print page.

### LINKED ISSUE #418: Replace hasRole calls with policies

### PR DESCRIPTION
**Pull request by luksan47**
Using policies where we already had them.

Fixes #418 
Fixes #359

### CODE REVIEW

File: app/Http/Controllers/InternetController.php [hunk 1]
```diff
@@ -149,16 +153,14 @@ public static function extendUsersInternetAccess(User $user)
...
         $validator = Validator::make($request->all(), [
             'comment' => 'required|max:1000',
             'mac_address' => ['required', 'regex:/((([a-fA-F0-9]{2}[-:]){5}([a-fA-F0-9]{2}))|(([a-fA-F0-9]{2}:){5}([a-fA-F0-9]{2})))/i'],
         ]);
         $validator->validate();
 
-        if ($validator->fails()) {
-            return back()->withErros($validator)->withInput();
-        }
-
```
Comment (kdmnk) [inline app/Http/Controllers/InternetController.php:161]: Why did you delete this?

Comment (kdmnk) [reply app/Http/Controllers/InternetController.php:161]: And the others as well

Comment (luksan47) [reply app/Http/Controllers/InternetController.php:161]: It seems to me that `$validator->validate();` does the same, so if the validator would fails, we return there and would never go into this function. You can check by adding a `die();` call after the `$validator->validate();` line.

Comment (kdmnk) [reply app/Http/Controllers/InternetController.php:161]: Yes, but do validate returns the error messages and the input? I think it only returns a template error message. Can you check?

Comment (luksan47) [reply app/Http/Controllers/InternetController.php:161]: Yes, both were returned here.

**Review by kdmnk — APPROVED** [review]
